lib/rgeo/geographic/interface.rb

```diff
@@ -315,29 +299,23 @@ def simple_mercator_factory(opts = {})
       # for more details.
 
       def projected_factory(opts = {})
-        CoordSys.check!(:proj4)
         if (projection_factory = opts[:projection_factory])
           # Get the projection coordinate systems from the given factory
-          projection_proj4 = projection_factory.proj4
-          unless projection_proj4
-            raise ArgumentError, "The :projection_factory does not have a proj4."
-          end
           projection_coord_sys = projection_factory.coord_sys
+          # TODO: may have to change to just CoordinateSystem because the proj4 implementation doesn't subclass ProjectedCoordinateSystem.
```

Comment:
Not sure here

Reply:
I think so. Proj4 just implements one class that can be any type of coordinate system (projected or geographic). But it has methods like `geographic?` that can be used to tell what type it is.

We should just add `geographic?` and `projected?` to the base `CS::CoordianteSystem` definition and children so we can rely on those.